docs/_tutorials/tcp-routes-mqtt.md

```diff
@@ -0,0 +1,406 @@
+ 
+```java
+@RequestMapping(value = "/message", method = RequestMethod.GET)
+public ResponseEntity<SimpleMessage> getLastMessageReceived() {
+    // ...
+}
+```
+
+The subscription JSON document used by the `/subscription` endpoint is modeled by the `SimpleSubscription` class, whereas the `/message` endpoint JSON document is modeled by the `SimpleMessage` class.
+
+For more details on sending and receiving messages, you can checkout the [MQTT Publish/Subscribe tutorial]({{ site.links-mqtt-pubsub-tutorial }}){:target="_top"}.
+
+## Building
+
+The full source code for this example is available in [GitHub]({{ site.repository }}){:target="_blank"}. To build, just clone and use gradle. Here is an example:
+
+```
+git clone {{ site.repository }}
+cd {{ site.baseurl | remove: '/'}} 
+./gradlew build
+```
+
+## Cloud Foundry Setup
+
+The sample application has a dependency on a `SERVICE_KEY` which needs to be satified by the user. A service key can be created for an existing Solace Messaging Service.
+
+### Creating the service
+
+Assuming TCP Routes is enabled and MQTT Plain-Text is set to "Enabled by default", you can create your service like so:
+
+```
+cf create-service solace-messaging shared solace-messaging-sample-instance
+```
+
+Assuming TCP Routes is enabled and MQTT Plain-Text is set to "Disabled by default", you can request MQTT be enabled upon service creation like so:
+
+```
+cf create-service solace-messaging shared solace-messaging-sample-instance -c '{ "mqtt_tcp_route_enabled" : "true" }'
+
+```
+
+### Creating the service key
+
+
+```
+cf create-service-key solace-messaging-sample-instance solace-messaging-sample-service-key
+cf service-key solace-messaging-sample-instance solace-messaging-sample-service-key 
+```
+
+## Getting and passing the service key to an application
+
+The service key details can be obtained from cf and should be passed to an application somehow. 
+When you have access to cf, get the service key and save it to a file so you may keep it and send it to where you plan on running your application.
+
+```
+cf service-key solace-messaging-sample-instance solace-messaging-sample-service-key | grep -v Getting > solace-messaging-sample-service.key
+```
+
+You need to get the file 'solace-messaging-sample-service.key' to where your application will be running somehow.
+
+Assuming you have the file now accessible where will run your application, load it in an environment variable we have chosen and named SERVICE_KEY.
+Load the service key from file to the SERVICE_KEY as an environment variable:
+
+```
+export SERVICE_KEY=$( cat solace-messaging-sample-service.key )
+```
+
+## Running
+
+This application can be used Standalone or deployed in PCF.

```

Comment:
Somewhere in the tutorial, want to explain when to use each option I think. Or perhaps don't bother with Option 2 since that is basically all the other tutorials in a nut shell isn't it?